lib/src/commands/music.dart

```diff
@@ -0,0 +1,161 @@
+            node.play(
+              context.guild!.id,
+              track,
+              requester: context.member!.id,
+              channelId: context.channel.id
+            ).queue();
+          }
+
+          await respond(context, MessageBuilder.content(
+              'Playlist `${result.playlistInfo.name}`($query) enqueued'
+          ));
+
+        } else {
+          node.play(
+            context.guild!.id,
+            result.tracks[0],
+            requester: context.member!.id,
+            channelId: context.channel.id
+          ).queue();
+          await respond(context, MessageBuilder.content(
+              'Track `${result.tracks[0].info?.title}` enqueued'
+          ));
+        }
+      })
+    ),
+    ChatCommand(
+      'skip',
+      'Skips the currently playing track',
+      checks: [connectedToAVoiceChannelCheck],
+      id('music-skip', (IChatContext context) async {
+        final node = MusicService.instance.cluster.getOrCreatePlayerNode(context.guild!.id);
+        final player = node.players[context.guild!.id]!;
+
+        if (player.queue.isEmpty) {
+          await respond(context, MessageBuilder.content('The queue is clear!'));
+          return;
+        }
+
+        node.skip(context.guild!.id);
+        await respond(context, MessageBuilder.content('Skipped current track'));
+      })
+    ),
+    ChatCommand(
+      'stop',
+      'Stops the current player and clears its track queue',
+      checks: [connectedToAVoiceChannelCheck],
+      id('music-stop', (IChatContext context) async {
+        final node = MusicService.instance.cluster.getOrCreatePlayerNode(context.guild!.id);
+        node.stop(context.guild!.id);
+        await respond(context, MessageBuilder.content('Player stopped!'));
+      })
+    ),
+    ChatCommand(
+      'leave',
+      'Leaves the current voice channel',
+      checks: [connectedToAVoiceChannelCheck],
+      id('music-leave', (IChatContext context) async {
+        (context.client as INyxxWebsocket)
+            .shardManager
+            .shards
+            .singleWhere((shard) => shard.guilds.contains(context.guild!.id))
+            .changeVoiceState(context.guild!.id, null);
+        await respond(context, MessageBuilder.content('Channel left'));
+      })
+    ),
+    ChatCommand(
+      'join',
+      'Joins the voice channel you are in',
+      checks: [notConnectedToAVoiceChannelCheck],
+      id('music-join', (IChatContext context) async {
+        MusicService.instance.cluster.getOrCreatePlayerNode(context.guild!.id);
+        await connectIfNeeded(context);
+        await respond(context, MessageBuilder.content('Joined your voice channel'));
+      })
+    ),
+    ChatCommand(
+      'volume',
+      'Sets the volume for the player',
+      checks: [connectedToAVoiceChannelCheck],
+      id('music-volume', (
+          IChatContext context,
+          @Description('The new volume, this value must be contained between 0 and 1000') int volume
```

Comment:
Could you also use the `@UseConverter(IntConverter(min: 0, max: 1000))` so that people using slash commands get a client-side hint on Mon and max values?

Reply:
done